### PR TITLE
Chem buffs

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Oxycodone and synap stuns on metabolism removed. They're replaced with a 60-second timer preventing microdose spamming of the instant staminaloss regen. Yay! Marines got a buff!

Inaprov buffed. Upon being injected with 14 or more units of inaprov while in crit, *instantly* heal 30% of your brute and burn damage, then go on cooldown for 300 seconds. Additionally, if you have greater than 29 points of organ damage and meet the same conditions (in crit, no cooldown), it will immediately heal enough organ damage to set your organ damage to 29.

Russian Red and Neuraline do the same thing, but with a somewhat weaker effect (20% heal, set organ damage to 29 points if higher than that, minimum volume requirements of  a standard dose minus 1u), and the same 300 second cooldown.

As this requires you to be in hardcrit, someone else must inject you, you *cannot* self-administer. It is additionally considered "free" heals and exempt from side effects.

Yes, if you spend points on this as a medic, you can make a one-shot injector that can straight-up insta-heal 65% of someone's physical damage. 
As long as they're in crit.
Once every 5 minutes.

## Why It's Good For The Game

Removes unneeded stuns that could be handled better as a timer. Also, encourages teamplay by making three semi-rare medicines work much better, but only if they're applied to someone *else* who's downed.

## Changelog
:cl:
del: Removed stun on oxy and synap expiration, replaced with a 10-second timer.
balance: If you are in crit and are given a dose of inaprov, you will heal 30% of your physical damage instantly and set any organs with more than 15 damage to 15 damage. Russian Red and Neuraline do the same thing, but 20% and 29 damage respectively. All three medicines have a 300-second cooldown before they'll perform the critheal function again.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
The stun removal is reasonable, but the instant heal seems pretty powerful, with no drawbacks. The 5 minute timer isn't that major either. 5 minutes between getting KO'd is not very long.

Also does that mean you can punch someoje in the face to edge them into crit then insta heal them back out?

**Comment by jroinc** [comment]
> 
> 
> The stun removal is reasonable, but the instant heal seems pretty powerful, with no drawbacks. The 5 minute timer isn't that major either. 5 minutes between getting KO'd is not very long.
> 
> Also does that mean you can punch someone in the face to edge them into crit then insta heal them back out?

It is fairly powerful, and I agree. There are no drawbacks for a few reasons (specific per drug and some general). 

The inaprov-specific one is that no one carries inaprovaline. It's been entirely subsumed by other drugs to the point that it's dedicated function (keeping you alive in crit) is seen as a *negative*. This gives it a use and might see someone carry it.

The neuraline-specific one is that it's a medic-only drug designed as a top-tier stimulant to drag people kicking and screaming out of crit. At... 15? points per three use injector, I'm cool with it having extra functionality.

The RR-specific one is that it will still deal cloneloss, which is punishment enough. I'd like to throw in some extra healing if you actually need it, and this fits the bill.

General reasons are that this encourages teamwork as you can't self-dose, and encourages trying to reach crit people before they fully die even harder. The 5-minute cooldown prevents players from just relying on it to get them through everything, while allowing a "comeback mechanism" the first time they go down.

In regards to punching someone into crit then healing them... I guess? That seems like you'd need to be in a very narrow window of HP for that to make sense, and as you can't know if the person's in the 5 minute window from a previous treatment, that seems kinda risky. I'd treat it the same as people who kill themselves cause they're hugged and want more time before they pop, tbh.

What would you change if you could?

**Comment by Lumipharon** [comment]
I definitely agree regarding teamwork, as half the marines have their own pharmacy with them at the moment for self reliance.

RR and neuraline buffs are good as well, but with the easily available inaprov I'm just a little bit wary of things that make getting back into the fight after getting killed/critted even faster, as that makes it tough for the benos.

The actual concept is real cool though, and being able to get a downed buddy on his feet with an injector is neat.

Actually one thing that you could do that would keep that effect but stop the medic speed run part... make inaprov give you a rav endure like effect when injected into a crit marine- you're health is past crit, but you can bounce back onto your feet for a while.  That changes the point of the drug, but as you said it's not used currently.

**Comment by Losenis** [comment]
As nice as it sounds I'd be wary about making it more difficult for antichem xeno castes to deal with marines even in crit. Last thing I'd want as defiler for example is the marine I managed to crit with a quadrillion neurotox but got chased off of by the other marines just immediately getting up despite the neurotox due to the chem metabolizing immediately for the upfront healing.

**Comment by jroinc** [comment]
> 
> 
> I definitely agree regarding teamwork, as half the marines have their own pharmacy with them at the moment for self reliance.
> 
> RR and neuraline buffs are good as well, but with the easily available inaprov I'm just a little bit wary of things that make getting back into the fight after getting killed/critted even faster, as that makes it tough for the benos.
> 
> The actual concept is real cool though, and being able to get a downed buddy on his feet with an injector is neat.
> 
> Actually one thing that you could do that would keep that effect but stop the medic speed run part... make inaprov give you a rav endure like effect when injected into a crit marine- you're health is past crit, but you can bounce back onto your feet for a while. That changes the point of the drug, but as you said it's not used currently.

Hmm. I'd have to look at how the effect is handled per rav, but I think that's a worthwhile idea. No promises tho.

**Comment by jroinc** [comment]
> 
> 
> As nice as it sounds I'd be wary about making it more difficult for antichem xeno castes to deal with marines even in crit. Last thing I'd want as defiler for example is the marine I managed to crit with a quadrillion neurotox but got chased off of by the other marines just immediately getting up despite the neurotox due to the chem metabolizing immediately for the upfront healing.

It only touches physical damage (brute/burn). Toxin, oxyloss, and stamina are unaffected, so if you crit someone with a million neurotox, he'll stay down.

**Comment by Losenis** [comment]
Neurotox also purges medicines, so if the guy gets in crit with a million neurotox the medicines will get purged. They'll work a bit, but hardly. Maybe to keep him alive, but not enough to actually heal him before getting purged, but this can give upfront healing regardless of metab rate or purging chems.

**Comment by jroinc** [comment]
> 
> 
> Neurotox also purges medicines, so if the guy gets in crit with a million neurotox the medicines will get purged. They'll work a bit, but hardly. Maybe to keep him alive, but not enough to actually heal him before getting purged, but this can give upfront healing regardless of metab rate or purging chems.

Yes, but only if he's in crit *due to physical damage*. If he's in crit due to neurotox or whatever chems, it'll still be untouched.

If he's in crit due to physical damage but then somehow got a ton of neuro/random ayy chems in him, this lets you save him... but he still has the ayychems and will still be subject to their effects, and will go right back into crit when the ayychems have their effects.

**Comment by jroinc** [comment]
Functional and performs as expected in testing. An inaprov pill given to a person beat barely into crit brought them into bright-red territory, but no longer flashing, and reset their appendix from 60 to 15 damage.

File: code/modules/reagents/reagents/medical.dm [hunk 1]
```diff
@@ -16,6 +16,25 @@
...
 
+/datum/reagent/medicine/inaprovaline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 14) //If you are in crit, and someone injects at least 15u into you at once, you will heal 30% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:27]: useless vars

File: code/modules/reagents/reagents/medical.dm [hunk 2]
```diff
@@ -16,6 +16,25 @@
...
+		L.adjustBruteLoss(-brute_loss * 0.30)
+		L.adjustFireLoss(-fire_loss * 0.30)
+		L.jitter(5)
+		for(var/datum/internal_organ/I in H.internal_organs)
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:31]:
```suggestion
		for(var/datum/internal_organ/I AS in H.internal_organs)
```
Ping me so i explain to you the difference

File: code/modules/reagents/reagents/medical.dm [hunk 3]
```diff
@@ -16,6 +16,25 @@
 	scannable = TRUE
 	trait_flags = TACHYCARDIC
 
+/datum/reagent/medicine/inaprovaline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 14) //If you are in crit, and someone injects at least 15u into you at once, you will heal 30% of your physical damage instantly.
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:24]: this sould not be 0.5, or even a ratio.

IIRC, human have a crit value somewhere, use it

File: code/modules/reagents/reagents/medical.dm [hunk 4]
```diff
@@ -444,6 +467,25 @@
...
 
+/datum/reagent/medicine/neuraline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 3) //If you are in crit, and someone injects at least 3u into you, you will heal 20% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:478]: useless vars

File: code/modules/reagents/reagents/medical.dm [hunk 5]
```diff
@@ -444,6 +467,25 @@
...
+		L.adjustBruteLoss(-brute_loss * 0.20)
+		L.adjustFireLoss(-fire_loss * 0.20)
+		L.jitter(10)
+		for(var/datum/internal_organ/I in H.internal_organs)
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:482]:
```suggestion
		for(var/datum/internal_organ/I AS in H.internal_organs)
```

File: code/modules/reagents/reagents/medical.dm [hunk 6]
```diff
@@ -517,10 +559,29 @@
...
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 9) //If you are in crit, and someone injects at least 9u into you, you will heal 20% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
+		L.adjustBruteLoss(-brute_loss * 0.20)
+		L.adjustFireLoss(-fire_loss * 0.20)
+		L.jitter(10)
+		for(var/datum/internal_organ/I in H.internal_organs)
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:578]:
```suggestion
		for(var/datum/internal_organ/I AS in H.internal_organs)
```

File: code/modules/reagents/reagents/medical.dm [hunk 7]
```diff
@@ -517,10 +559,29 @@
...
 
+/datum/reagent/medicine/russian_red/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 9) //If you are in crit, and someone injects at least 9u into you, you will heal 20% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:574]: vars

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — CHANGES_REQUESTED** [review]
LGTM otherwise

File: code/modules/reagents/reagents/medical.dm (hunk 3 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:24]: I checked and I think the crit value is literally just -0.5 *max health.

File: code/modules/reagents/reagents/medical.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:27]: What should it be instead?

File: code/modules/reagents/reagents/medical.dm [hunk 8]
```diff
@@ -16,6 +16,25 @@
...
+/datum/reagent/medicine/inaprovaline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 14) //If you are in crit, and someone injects at least 15u into you at once, you will heal 30% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
+		L.adjustBruteLoss(-brute_loss * 0.30)
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:28]:
```suggestion
		L.adjustBruteLoss(-L.getBruteLoss() * 0.30)
```

**Review by BraveMole — CHANGES_REQUESTED** [review]
try that

File: code/modules/reagents/reagents/medical.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 2 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 3 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 4 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 5 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 6 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 7 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

File: code/modules/reagents/reagents/medical.dm (hunk 8 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm]: done

**Comment by jroinc** [comment]
did the things

**Review by BraveMole — APPROVED** [review]

**Comment by Lumipharon** [comment]
I've been playing medic a lot with these changes, and they're all great (from a medic's point of view) dunno about balance though, as that's pretty hard to quantify.

The inaprov change definitely makes reviving/pulling people out of crit far easier, so marine turn around time is way faster. This might make it a bit tougher for xenos, as anyone retrieved is back in action fast than ever.
The organ healing kinda makes organ damage irrelevant tbh, and compounds with the above. And if your other PR that causes organ damage under 30 to slowly heal on it's on goes through, it will be even more negligible.

With the synap change in particular, I'm able to run nonstop by just dosing up on a stim hypo, which is kinda funny when I'm in heavy armour + Tyr. Even with the loss of the insta stam recovery, the per tick recovery is so high I can just dose a single unit, wait for it to expire and redose to refresh the timer thing.
The stun did make it shit though so removing it is good, but dunno how to balance this properly outside of properly controlling access to it.
With the apparent ever easier access to chemistry we're seeing it means we'll get to the stage pretty quickly where lots of people realise they can do this in like 10 seconds, and just run down xenos even easier. As a xeno player, having marines outrun you is always pretty horrible.
